apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -67,11 +67,22 @@ func (r *OpenTelemetryCollector) Default() {
 		r.Spec.TargetAllocator.Replicas = &one
 	}
 
-	if r.Spec.MaxReplicas != nil {
+	if r.Spec.MaxReplicas != nil || (r.Spec.Autoscaler != nil && r.Spec.Autoscaler.MaxReplicas != nil) {
 		if r.Spec.Autoscaler == nil {
 			r.Spec.Autoscaler = &AutoscalerSpec{}
 		}
 
+		if r.Spec.Autoscaler.MaxReplicas == nil {
+			r.Spec.Autoscaler.MaxReplicas = r.Spec.MaxReplicas
+		}
+		if r.Spec.Autoscaler.MinReplicas == nil {
+			if r.Spec.MinReplicas != nil {
+				r.Spec.Autoscaler.MinReplicas = r.Spec.MinReplicas
+			} else {
+				r.Spec.Autoscaler.MinReplicas = r.Spec.Replicas
```

Comment:
Are we sure that this is correct?

Reply:
Hmm I believe the logic is correct, but maybe there's something I'm missing?

`MinReplicas` is meant to be an optional argument, but it should eventually always have a value if `MaxReplicas` is set. So here it defaults to `.Spec.Replicas` which is validated by this [check](https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1alpha1/opentelemetrycollector_webhook.go#L158) in the validator. I'm setting this default earlier now instead of waiting for the creation of the autoscaler to set the default [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/collector/horizontalpodautoscaler.go#L83).

Wondering your thoughts and why this might not be the desired logic?

Reply:
> Wondering your thoughts and why this might not be the desired logic?

Originally I thought that the idea here is to totally decouple these two.  But it's reasonable to default to the `.spec.replicas`